controllers/clustercurator_controller.go

```
@@ -83,6 +85,27 @@ func (r *ClusterCuratorReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		return ctrl.Result{}, err
 	}
 
+	// Hypershift clusters need additional RBAC
+	if curator.Name != curator.Namespace {
+		log.V(2).Info("Check if cluster namespace " + curator.Name + " exists")
+		if _, err := r.Kubeset.CoreV1().Namespaces().Get(context.TODO(), curator.Name, v1.GetOptions{}); err != nil {
```

only if the err is not found otherwise might need to return the err